### DISCUSSION TITLE
'UGrid' object has no attribute 'compute_normals'

### DISCUSSION THREAD

**Discussion by RyanConway91** [marcomusy/vedo#725] [Q&A] [answered]
Hello, I am trying to intersect a mesh loaded from a vtk (a 3D irregular rectilinear grid from a finite difference model, 'mf_grid' variable below) with a wedge mesh loaded from some other commercial software (EVS, variable 'ugrid' below). I got both loaded in and they display correctly. 

However, when I run:

`grid_int = mf_grid.boolean(operation ='intersect',mesh2 = ugrid)`

I get the error:

`'UGrid' object has no attribute 'compute_normals'`

Can I convert my meshes to structured using vedo or is there some way I can get around this?

Thanks

**Comment by marcomusy** [comment, marked as answer]
Sorry for the late reply, I'm afraid vedo does not support data structures like rectilinear grids (maybe in the future...) I suggest to explore a [similar library](https://github.com/pyvista/pyvista) which is also based on vtk and it is largely compatible with vedo.
If you just need to do visualization you can still make a mesh from the vtk (as you've done already) and then intersect those meshes:
```python
import vedo
rgrid = vedo.io.loadRectilinearGrid("file.vtk")
msh1 = vedo.geometry(rgrid)
msh1.print()
msh1.cut_with_mesh(msh2)
vedo.show(msh1, msh2, axes=1)
```

**Reply by RyanConway91** [reply]
Your responsiveness has been amazing, no worries! 

Ugh so I got my meshes into pyvistas, but I get what seems like the same issue as I got in vedo (error in pyvistas is 'UnstructuredGrid' object has no attribute 'boolean_intersection')

Is there something fundamental about an unstructured grid that makes these types of operations impossible (I have a rectilinear mesh and wedge mesh)? 

Is there a way to convert these meshes to 'structured'? I ran both of the meshes through triangulate (in vedo and pyvistas) but I still cant intersect. I tried running them through "tetgen" (https://github.com/pyvista/tetgen) as well, but got an error on that one...

Seems like this should be a fairly common task in 3D computing. Am I missing an obvious solution or  thinking about this wrong?

**Reply by marcomusy** [reply]
If i rember correctly the method is called clip() to cut a grid to output an unstructured grid but you will need to check the pyvista docs.
The boolean operation is only working with polygonal meshes.
Cc @banesullivan

**Reply by banesullivan** [reply]
In PyVista, the boolean operations are only available on the `PolyData` type. You may need to run the `extract_surface()` filter in PyVista to get a `PolyData` from an `UnstructuredGrid`.

Sorry for my delay!! Hope this helps!
